crates/bitcoind_rpc/src/bip158.rs

```diff
@@ -31,23 +35,32 @@ pub struct FilterIter<'c, C> {
     spks: Vec<ScriptBuf>,
     // local cp
     cp: Option<CheckPoint>,
-    // blocks map
-    blocks: BTreeMap<Height, BlockHash>,
+    // block headers
+    headers: BTreeMap<Height, HashedHeader>,
+    // heights of matching blocks
+    matched: BTreeSet<Height>,
     // best height counter
     height: Height,
+    // initial height
+    start: Height,
     // stop height
     stop: Height,
 }
 
 impl<'c, C: RpcApi> FilterIter<'c, C> {
+    /// Hard cap on how far to walk back when a reorg is detected.
+    const MAX_REORG_DEPTH: u32 = 100;
```

Comment:
I understand that the main motivation for this is to protect against attacks from a malicious/faulty node?

This is not going to work as BDK does not check PoW (not currently). The node can just cheaply construct multiple <100 block reorgs and exhaust resources that way.